### PR TITLE
layers: Add VK_KHR_dynamic_rendering VUs and Tests

### PR DESCRIPTION
**Pull request by AaronHaganAMD**
VUs Added:
VUID-VkCommandBufferBeginInfo-flags-00053
VUID-VkGraphicsPipelineCreateInfo-flags-06482
VUID-VkGraphicsPipelineCreateInfo-flags-06483
VUID-VkGraphicsPipelineCreateInfo-renderPass-06051
VUID-VkGraphicsPipelineCreateInfo-renderPass-06063

Tests Added:
VUID-VkCommandBufferBeginInfo-flags-00053
VUID-VkCommandBufferBeginInfo-flags-00054
VUID-VkCommandBufferBeginInfo-flags-06000
VUID-VkCommandBufferBeginInfo-flags-06001
VUID-VkCommandBufferBeginInfo-flags-06002
VUID-VkGraphicsPipelineCreateInfo-flags-06482
VUID-VkGraphicsPipelineCreateInfo-flags-06483
VUID-VkGraphicsPipelineCreateInfo-renderPass-06051
VUID-VkGraphicsPipelineCreateInfo-renderPass-06063

### CODE REVIEW

**Review by ncesario-lunarg — COMMENTED** [review]
Some comments/questions that might need to get resolved, but generally LGTM.

I'm wondering if it's possible to move some of these to stateless validation. e.g., looks like 06482, 06483, 06063 only depends on the CreateInfo passed in 
 at CreateGraphicsPipelines time? If so, these checks should probably be moved to `StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines`.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5756,7 +5798,12 @@ bool CoreChecks::PreCallValidateCreateGraphicsPipelines(VkDevice device, VkPipel
 
     for (uint32_t i = 0; i < count; i++) {
         if (pCreateInfos[i].renderPass == VK_NULL_HANDLE) {
-            if (!enabled_features.core13.dynamicRendering) {
+            if ((api_version < VK_API_VERSION_1_3) && (!enabled_features.core13.dynamicRendering) &&
+                (pCreateInfos[i].renderPass == VK_NULL_HANDLE)) {
+                skip |= LogError(device, "VUID-VkGraphicsPipelineCreateInfo-renderPass-06051",
+                                 "vkCreateGraphicsPipelines() pCreateInfo[%" PRIu32 "]: renderpass must not be VK_NULL_HANDLE.", i);
+                return true;
```
[the remark this answers is not in the document]
Comment (Tony-LunarG) [reply layers/core_validation.cpp:5805]: The return true came from "layers: Avoid crashing if VK_KHR_dynamic_rendering is enabled" but whether it's still needed or not I couldn't say.

File: tests/vklayertests_dynamic_rendering.cpp [hunk 2]
```diff
@@ -1064,3 +1084,124 @@ TEST_F(VkLayerTest, DynamicRenderingAttachmentInfo) {
...
+         VK_ATTACHMENT_LOAD_OP_DONT_CARE, VK_ATTACHMENT_STORE_OP_DONT_CARE, VK_IMAGE_LAYOUT_UNDEFINED,
+         VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL},
+    };
+    VkAttachmentReference att_ref = {0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL};
+
+    VkSubpassDescription subpass = {0, VK_PIPELINE_BIND_POINT_GRAPHICS, 0, nullptr, 1, &att_ref, nullptr, nullptr, 0, nullptr};
+    VkRenderPassCreateInfo rpci = {VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO, nullptr, 0, 1, attach, 1, &subpass, 0, nullptr};
+    
+    VkRenderPass rp1;
+    auto ret = vk::CreateRenderPass(m_device->device(), &rpci, nullptr, &rp1);
```
Comment (Tony-LunarG) [inline tests/vklayertests_dynamic_rendering.cpp:1148]: rp1 needs to be destroyed at the end of the test.  This is true in the next test too.

Comment (jeremyg-lunarg) [reply tests/vklayertests_dynamic_rendering.cpp:1148]: If possible, please use vk_testing::RenderPass to get RAII cleanup of the handle.

**Comment by AaronHaganAMD** [comment]
> Some comments/questions that might need to get resolved, but generally LGTM.
> 
> I'm wondering if it's possible to move some of these to stateless validation. e.g., looks like 06482, 06483, 06063 only depends on the CreateInfo passed in at CreateGraphicsPipelines time? If so, these checks should probably be moved to `StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines`.

I will refactor the stateless tests to manual_PreCallValidateCreateGraphicsPipelines.

File: layers/parameter_validation_utils.cpp [hunk 3]
```diff
@@ -2062,6 +2062,52 @@ bool StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines(VkDevice
                                  i);
             }
 
+            if (pCreateInfos[i].pColorBlendState) {
```
Comment (ncesario-lunarg) [inline layers/parameter_validation_utils.cpp:2065]: So the test failures are due to this (and similar pointers) needing to be ignored under certain conditions. These conditions are automatically checked due to how the CreateInfo struct is constructed in PIPELINE_STATE.

In general I think we want to move any checks that _can_ happen in stateless validation outside of core, but that would require extra work here. @Tony-LunarG @jeremyg-lunarg @aitor-lunarg, do you think it's worth doing some refactoring to more easily allow these pointers to be checked at parameter validation time, or rather just leave these checks in core for now?

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**